src/test/regress/sql/multi_cross_shard.sql

```diff
@@ -0,0 +1,64 @@
+--
+-- MULTI_CROSS_SHARD
+--
+-- Tests to log cross shard queries according to error log level
+--
+
+-- Create a distributed table and add data to it
+CREATE TABLE multi_task_tt(id int, name varchar(20));
+SELECT create_distributed_table('multi_task_tt', 'id');
+INSERT INTO multi_task_tt VALUES(1, 'elem_1');
+INSERT INTO multi_task_tt VALUES(2, 'elem_2');
+INSERT INTO multi_task_tt VALUES(3, 'elem_3');
+
+-- Shouldn't log anything when the log level is 'off'
+SHOW citus.multi_task_query_log_level;
+SELECT * FROM multi_task_tt;
+
+-- Get messages with the log level 'notice'
+SET citus.multi_task_query_log_level TO notice;
+SELECT * FROM multi_task_tt;
+SELECT AVG(id) AS avg_id FROM multi_task_tt;
+
+-- Get messages with the log level 'error'
+SET citus.multi_task_query_log_level TO error;
+SELECT * FROM multi_task_tt;
+
+-- Check the log message with INSERT INTO ... SELECT 
+CREATE TABLE raw_table
+(
+    id int,
+    order_count int
+);
+
+CREATE TABLE summary_table
+(
+    id int,
+    order_sum BIGINT
+);
+
+SELECT create_distributed_table('raw_table', 'id');
+SELECT create_distributed_table('summary_table', 'id');
+
+INSERT INTO raw_table VALUES(1, '15');
+INSERT INTO raw_table VALUES(2, '15');
+INSERT INTO raw_table VALUES(3, '15');
+INSERT INTO raw_table VALUES(1, '20');
+INSERT INTO raw_table VALUES(2, '25');
+INSERT INTO raw_table VALUES(3, '35');
+
+--  Should notice user that the query is multi-task one
+SET citus.multi_task_query_log_level TO notice;
+INSERT INTO summary_table SELECT id, SUM(order_count) FROM raw_table GROUP BY id;
+
+-- Should error out since the query is multi-task one
+SET citus.multi_task_query_log_level TO error;
+INSERT INTO summary_table SELECT id, SUM(order_count) FROM raw_table GROUP BY id;
+
+-- Should have only three rows
+SET citus.multi_task_query_log_level to DEFAULT;
+SELECT * FROM summary_table;
+
+DROP TABLE multi_task_tt;
+DROP TABLE raw_table;
+DROP TABLE summary_table;
```

Comment:
We should also add queries for task_tracker, router-select and single task insert-into select.